### PR TITLE
tests: Add SIGKILL functional test

### PR DESCRIPTION
**Pull request by serathius**
Splitting https://github.com/etcd-io/etcd/pull/13838 into mergable code
cc @ptabor @ahrtr

### CODE REVIEW

**Comment by serathius** [comment]
@ahrtr @ptabor PTAL

File: tests/functional/functional.yaml [hunk 1]
```diff
@@ -185,6 +185,8 @@ tester-config:
   - SIGTERM_ALL
   - SIGQUIT_AND_REMOVE_ONE_FOLLOWER
   - SIGQUIT_AND_REMOVE_ONE_FOLLOWER_UNTIL_TRIGGER_SNAPSHOT
+  - SIGKILL_FOLLOWER
```
Comment (ahrtr) [inline tests/functional/functional.yaml:188]: `SIGKILL_ONE_FOLLOWER` so as to keep the naming consistent as the existing `SIGTERM_ONE_FOLLOWER` and `SIGQUIT_AND_REMOVE_ONE_FOLLOWER` ?

File: tests/functional/rpcpb/rpc.proto [hunk 2]
```diff
@@ -426,6 +429,8 @@ enum Case {
   // each member must be able to process client requests.
   SIGQUIT_AND_REMOVE_QUORUM_AND_RESTORE_LEADER_SNAPSHOT_FROM_SCRATCH = 14;
 
+  SIGKILL_FOLLOWER = 15;
```
Comment (ahrtr) [inline tests/functional/rpcpb/rpc.proto:432]: SIGKILL_ONE_FOLLOWER? The same as above.

**Comment by ahrtr** [comment]
Basically looks good to me with one minor comment. 

Another comment is there are some duplicate code, such as [handler.go#L555-L565](https://github.com/etcd-io/etcd/blob/3a239f7b3b02964deb4f2dd4489856e01fbb9906/tests/functional/agent/handler.go#L555-L565). We may get them wrapped into a common function or method. But of course we can address it in a separate PR. I can do it either once this one is merged.

**Review by ptabor — APPROVED** [review]

**Review by ptabor — CHANGES_REQUESTED** [review]
nits, but marking as requiring changing for attention.

**Pull request closed without merging**